src/mad_spark_alt/core/retry.py

```
@@ -358,14 +358,15 @@ async def _make_request() -> Any:
-        except aiohttp.ClientTimeout:  # type: ignore
-            raise LLMTimeoutError("Request timed out")
+        except aiohttp.ClientTimeout as e:
+            raise LLMTimeoutError("Request timed out") from e
         except aiohttp.ClientError as e:
-            raise NetworkError(f"Network error: {str(e)}")
+            raise NetworkError(f"Network error: {str(e)}") from e
 
     # Apply circuit breaker if provided
     if circuit_breaker:
-        request_func = lambda: circuit_breaker.call(_make_request)
+        async def request_func():
```

Ruff (0.11.9)

Missing return type annotation for private function `request_func`

(ANN202)